resources/analysisTools/indelCallingWorkflow/checkSampleSwap_TiN.pl

```diff
@@ -294,21 +315,42 @@
 #######################################
 ### Finding and plotting TiN
 
-print "Rscript $TiN_R -f $snvsGT_germlineRare_txt --oPlot $snvsGT_germlineRare_png --oFile $snvsGT_germlineRare_oFile -p $pid --chrLength $chrLengthFile --cFunction $canopy_Function --SeqType $seqType --rightBorder $rightBorder --bottomBorder $bottomBorder --vcf $snvsGT_germlineRare --Ovcf $snvsGT_germlineRare_oVCF\n";
+my $runRscript_code  = "Rscript $TiN_R -f $snvsGT_germlineRare_txt";
+   $runRscript_code .= " --oPlot $snvsGT_germlineRare_png";
+   $runRscript_code .= " --oFile $snvsGT_germlineRare_oFile";
+   $runRscript_code .= " -p $pid";
+   $runRscript_code .= " --chrLength $chrLengthFile";
+   $runRscript_code .= " --cFunction $canopy_Function";
+   $runRscript_code .= " --SeqType $seqType";
+   $runRscript_code .= " --rightBorder $rightBorder";
+   $runRscript_code .= " --bottomBorder $bottomBorder";
+   $runRscript_code .= " --vcf $snvsGT_germlineRare";
+   $runRscript_code .= " --Ovcf $snvsGT_germlineRare_oVCF";
 
-my $runRscript = system("Rscript $TiN_R -f $snvsGT_germlineRare_txt --oPlot $snvsGT_germlineRare_png --oFile $snvsGT_germlineRare_oFile -p $pid --chrLength $chrLengthFile --cFunction $canopy_Function --SeqType $seqType --rightBorder $rightBorder --bottomBorder $bottomBorder --vcf $snvsGT_germlineRare --Ovcf $snvsGT_germlineRare_oVCF\n");
+print $runRscript_code, "\n";
+
+my $runRscript = system($runRscript_code);
 
 if($runRscript != 0) { 
   `rm $jsonFile`;
-  die "Error while running $TiN_R in swapChecker\n";
+  if($runRscript == 50) {
+    $! = 50;
```

Comment:
What is the reason to fill a variable with such a strange name '$!' with a value rather than directly writing the exit code in the message or to use '$runRscript' directly?

Reply:
better use `errcode` or so.